### PR TITLE
CLN: Removed coerce param in pd.to_timedelta and pd.to_datetime

### PR DESCRIPTION
**Pull request by gfyoung**
Deprecated back in `v0.17.0` <a href="https://github.com/pydata/pandas/commit/987b7e7e586b8df1d127406c69e0a9094a1a5322">here</a>.  Seems appropriate to carry though now.

### CODE REVIEW

File: doc/source/whatsnew/v0.19.0.txt
Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: add what it's in favor of

File: pandas/tseries/timedeltas.py
Comment (jreback) [inline pandas/tseries/timedeltas.py]: hmm no tests to remove?

Comment (gfyoung) [inline pandas/tseries/timedeltas.py]: Seems not.  That's on you! :smile:

File: doc/source/whatsnew/v0.19.0.txt
Comment (gfyoung) [inline doc/source/whatsnew/v0.19.0.txt]: Done.  Old build to cancel <a href="https://travis-ci.org/pydata/pandas/builds/143620635">here</a>.

**Comment by gfyoung** [comment]
@jreback : Travis is passing, so ready to merge if there are no other concerns.

**Comment by jreback** [comment]
actually, I don't think this can be done w/o doing for ``to_datetime`` as well. Either both or none.

**Comment by jreback** [comment]
and we generally are waiting much longer, see the list @jorisvandenbossche put up at the very bottom:
https://github.com/pydata/pandas/issues/6581 

any of those are fair game. Yes we could/should do some of the 0.17.0 ones, but let's get some of the others first.

**Comment by gfyoung** [comment]
@jreback : so if I also remove the `to_datetime` param, will this be merged in for `0.19.0` or not?  A duration of two major releases seems like a good amount of time for people to adjust their code.

**Comment by jreback** [comment]
@gfyoung yes it is *normally* ok, but we SO many deprecations (esp from 0.17.0.), that I think let's clean everything up thats < 0.17.0 first. So I would say defer this to 0.20.0..

**Comment by gfyoung** [comment]
@jreback : Hmm...I'm not sure I totally follow that logic, especially since I already do see a deprecation removal from `0.17.0` being listed in that issue that you linked to (the legacy offsets one).

**Comment by jreback** [comment]
yeah. I don't have a problem with anything from 0.17.0. BUT this is a 'biggie' (well to_datetime is really). So no objection if you include both (for consistency). But if you have a limited pool of PR energy, would prefer other ones to be addressed .....

**Comment by gfyoung** [comment]
@jreback : this one isn't that difficult to remove AFAICT - I'll also take a look at some of the other deprecation removals and put up PR's for those if I can.

**Comment by jreback** [comment]
right, add on ``to_datetime`` removal then would be ok, but @jorisvandenbossche might feel different.

**Comment by gfyoung** [comment]
@jreback : fair enough - in the meantime, I think we do have one deprecation removal that I think all of us can agree upon (#13419) :smile:

**Comment by gfyoung** [comment]
@jreback : removed the `coerce` parameter in `to_datetime`, and Travis is still passing. @jorisvandenbossche thoughts?

**Comment by gfyoung** [comment]
@jreback : removed the `coerce` parameter in `to_datetime`, and Travis is still passing. @jorisvandenbossche thoughts?

**Comment by jorisvandenbossche** [comment]
Personally I would table this for 0.20 .. 
The other deprecated things we are removing are all even from 0.15 and (the maybe's) from 0.16 (except for the offset/freq PR), and the 0.17 release is less than a year old.

**Comment by jorisvandenbossche** [comment]
Maybe we need some more 'formal' rules on how long deprecations are kept at the minimum? Eg minimal two feature releases and minimal a year.

**Comment by gfyoung** [comment]
@jorisvandenbossche: When has two feature releases ever taken more than a year?  Two feature releases sure, but a year seems a little long too me, and not just because I would like to merge this ;)

**Comment by gfyoung** [comment]
@jorisvandenbossche: When has two feature releases ever taken more than a year?  Two feature releases sure, but a year seems a little long too me, and not just because I would like to merge this ;)

**Comment by jreback** [comment]
I think let's defer this one to 0.20. I don't see any downside to delaying this.

**Comment by jorisvandenbossche** [comment]
> but a year seems a little long too me, and not just because I would like to merge this ;)

We may not like it, but it is a fact that not all users upgrade each version but often skip a few at once ... 
I understand you rather see it merged, but rebasing it in a few months will also not be that much work!

**Comment by gfyoung** [comment]
@jorisvandenbossche : this deprecation removal process is somewhat new to me, and I was not aware of user tendencies to skip version upgrades (I generally follow them quite closely for my projects).  However, I perfectly understand!  I was being somewhat facetious when I mentioned that I wanted to merge this now from the `:)` I added at the end.

**Comment by jreback** [comment]
@gfyoung as a further data point. We have tended to deprecate a lot, and only are now catching up with removals. I think 0.19/0.20 will be a good fix point for some of this.

File: doc/source/whatsnew/v0.19.0.txt
Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: this is prob causing conflicts i would take out for now

Comment (gfyoung) [inline doc/source/whatsnew/v0.19.0.txt]: Ah, good point.  Should move to the 0.20 `whatsnew`.

Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: yep. why don't you create a new 0.20 (in sep PR), then will be nicer.

Comment (gfyoung) [inline doc/source/whatsnew/v0.19.0.txt]: No need.  There already is one.

**Review by jorisvandenbossche — APPROVED** [review]
This looks OK to me!

**Comment by jorisvandenbossche** [comment]
@gfyoung can you force push again? (possibly due to the repo transfer the PRs open before the transfer cannot be merged from the github interface)

**Comment by jreback** [comment]
@jorisvandenbossche if you merge this, we for sure need to branch 0.19.1 (which is ok)

**Comment by jreback** [comment]
also should remove ``coerce`` from ``pd.to_datetime`` for consistency.

**Comment by gfyoung** [comment]
@jreback : Ah, good point.  I didn't realize it had been deprecated back in `0.17.0`.  Will do then.

**Comment by gfyoung** [comment]
@jreback : Actually, never mind!  I did remove it in this PR!  Need to change the title.

**Comment by jreback** [comment]
ok lgtm. our first non-0.19.1 PR!

**Comment by jorisvandenbossche** [comment]
Merged! Thanks @gfyoung

**Comment by jreback** [comment]
@jorisvandenbossche so need to branch 0.19.1 (as this should not be included)!

**Pull request closed without merging**
